mmhuman3d/utils/transforms.py

```diff
@@ -546,3 +546,144 @@ def _sja_to_aa(sja, R_t, R_t_inv):
+    T_v2p = numpy.eye(4).reshape(1, 4, 4).repeat(N, axis=0)
+    global_orient_mat = aa_to_rotmat(global_orient)
+    T_v2p[:, :3, :3] = global_orient_mat
+    T_v2p[:, :3, 3] = transl
+
+    # compute combined transformation from vertex to world
+    T_v2w = T_p2w @ T_v2p
+
+    # compute transformation from vertex to camera
+    T_v2c = T_w2c @ T_v2w
+
+    # decompose vertex to camera transformation
+    # np: new pelvis frame
+    # T_v2c = T_np2c x T_v2np
+    T_np2c = T_p2w
+    T_v2np = numpy.linalg.inv(T_np2c) @ T_v2c
+
+    # decompose into new global orient and new transl
+    new_global_orient_mat = T_v2np[:, :3, :3]
+    new_gloabl_orient = rotmat_to_aa(new_global_orient_mat)
+    new_transl = T_v2np[:, :3, 3]
+
+    assert new_gloabl_orient.shape == (N, 3)
+    assert new_transl.shape == (N, 3)
+
+    return new_gloabl_orient, new_transl
+
+
+def quat_feat(theta: torch.Tensor) -> torch.Tensor:
+    """Computes a normalized quaternion ([0,0,0,0]  when the body is in rest
+    pose) given joint angles.
+
+    Args:
+        theta (torch.Tensor): A tensor of joints axis angles,
+            batch size x number of joints x 3
+
+    Returns:
+        quat (torch.Tensor)
+    """
+    l1norm = torch.norm(theta + 1e-8, p=2, dim=1)
+    angle = torch.unsqueeze(l1norm, -1)
+    normalized = torch.div(theta, angle)
+    angle = angle * 0.5
+    v_cos = torch.cos(angle)
+    v_sin = torch.sin(angle)
+    quat = torch.cat([v_sin * normalized, v_cos - 1], dim=1)
+    return quat
+
+
+def _quat2mat(quat: torch.Tensor) -> torch.Tensor:
+    """Converts a quaternion to a rotation matrix.
+
+    Args:
+        quat (torch.Tensor)
+
+    Returns:
+        rotMat (torch.Tensor)
+    """
+    norm_quat = quat
+    norm_quat = norm_quat / norm_quat.norm(p=2, dim=1, keepdim=True)
+    w, x, y, z = norm_quat[:, 0], norm_quat[:, 1], norm_quat[:, 2], \
+        norm_quat[:, 3]
+    B = quat.size(0)
+    w2, x2, y2, z2 = w.pow(2), x.pow(2), y.pow(2), z.pow(2)
+    wx, wy, wz = w * x, w * y, w * z
+    xy, xz, yz = x * y, x * z, y * z
+    rotMat = torch.stack([
+        w2 + x2 - y2 - z2, 2 * xy - 2 * wz, 2 * wy + 2 * xz, 2 * wz + 2 * xy,
+        w2 - x2 + y2 - z2, 2 * yz - 2 * wx, 2 * xz - 2 * wy, 2 * wx + 2 * yz,
+        w2 - x2 - y2 + z2
+    ],
+                         dim=1).view(B, 3, 3)
+    return rotMat
+
+
+def rodrigues(theta: torch.Tensor) -> torch.Tensor:
+    """Computes the rodrigues representation given joint angles.
+
+    Parameters
+    ----------
+    :param theta: batch_size x number of joints x 3
+    :return: batch_size x number of joints x 3 x 4
+    """
+    l1norm = torch.norm(theta + 1e-8, p=2, dim=1)
+    angle = torch.unsqueeze(l1norm, -1)
+    normalized = torch.div(theta, angle)
+    angle = angle * 0.5
+    v_cos = torch.cos(angle)
+    v_sin = torch.sin(angle)
+    quat = torch.cat([v_cos, v_sin * normalized], dim=1)
+    return _quat2mat(quat)
+
+
+def with_zeros(input: torch.Tensor) -> torch.Tensor:
```

Comment:
Nice function, two suggestions:
1. `with_zeros` might be unclear. It may be better called something along the line of `make_homegeneous_rotmat`
2. Can we make it more general that it also accepts a single transformation matrix (i.e. shape is (3, 4), no batch dim)